expression/schema.go

```diff
@@ -104,10 +129,12 @@ func (s *Schema) Append(col *Column) {
 
 // MergeSchema will merge two schema into one schema.
 func MergeSchema(lSchema, rSchema Schema) Schema {
-	return NewSchema(append(lSchema.Clone().Columns, rSchema.Clone().Columns...))
+	tmpL := lSchema.Clone()
+	tmpR := rSchema.Clone()
+	return NewSchema(append(tmpL.Columns, tmpR.Columns...), append(tmpL.Keys, tmpR.Keys...))
 }
 
 // NewSchema returns a schema made by its parameter.
-func NewSchema(cols []*Column) Schema {
-	return Schema{Columns: cols}
+func NewSchema(cols []*Column, keys []KeyInfo) Schema {
```

Comment:
It's no need to add it to constructor. Just add a function SetUniqueKey(...*KeyInfo)